File: T1/Etapa4/src/Publisher.java
public class Publisher extends Component {
    private Broker broker;
    private String topic;

    public Publisher(String name, Broker broker, String topic) {
        super(name);
        this.broker = broker;
        this.topic = topic;
    }

    public void publish(String message) {
        broker.publish(topic, name, message);
    }
}

File: T3/Entrega/publisher.cpp
#include "publisher.h"

Publisher::Publisher(const QString& name, Topic* topic) 
    : Component(name, topic) {}

void Publisher::publish(const QString& message) {
    notifySubscribers(message);
}

void Publisher::notifySubscribers(const QString& message) {
    if (topic) {
        topic->notifySubscribers(message);
    }
}
